pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -270,7 +270,11 @@ func (r *Builder) getSourceDetails(vm *model.VM, sourceSecret *core.Secret) (lib
 			Path:     "",
 			RawQuery: sslVerify,
 		}
-		fingerprint = host.Thumbprint
+		if r.Source.Provider.Spec.Settings[api.SDK] == api.ESXI {
```

Comment:
we have this condition below.
what's the difference between `hostDef, found := r.hosts[host.ID]; found` to `r.Source.Provider.Spec.Settings[api.SDK] == api.ESXI`?
if here we can get the same condition?
also we use `fingerprint = r.Source.Provider.Status.Fingerprint` in every if case, the only different case is when `r.Source.Provider.Spec.Settings[api.SDK] != api.ESXI`. maybe we can set it in a single place and then only change it in this case?

Reply:
> what's the difference between `hostDef, found := r.hosts[host.ID]; found` to `r.Source.Provider.Spec.Settings[api.SDK] == api.ESXI`? if here we can get the same condition?

see the `Host` part in the URL - when there's an ESXi network, we set it to `hostDef.Spec.IpAddress` and when there's no ESXi network, we set it to `host.Name`, and the source of the credentials also changes

> also we use `fingerprint = r.Source.Provider.Status.Fingerprint` in every if case, the only different case is when `r.Source.Provider.Spec.Settings[api.SDK] != api.ESXI`. maybe we can set it in a single place and then only change it in this case?

that makes less sense - it would make the code less expressive